Fix zerorpc tests, move them to separate test env

ramshaw888 opened the pull request:
zerorpc tests are currently being skipped all of the time because of an incorrect gevent version specified. The line `zerorpc = pytest.importorskip("zerorpc")` actually raises an `ImportError` on it's dependency `gevent`, NOT on `zerorpc` itself:

```
___________________________________________ ERROR collecting tests/contrib/zerorpc/zeropc_tests.py ___________________________________________
ImportError while importing test module '/app/tests/contrib/zerorpc/zeropc_tests.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python2.7/site-packages/six.py:709: in exec_
    exec("""exec _code_ in _globs_, _locs_""")
tests/contrib/zerorpc/zeropc_tests.py:40: in <module>
    import zerorpc
/usr/local/lib/python2.7/site-packages/zerorpc/__init__.py:28: in <module>
    from .socket import *
/usr/local/lib/python2.7/site-packages/zerorpc/socket.py:27: in <module>
    from .events import Events
/usr/local/lib/python2.7/site-packages/zerorpc/events.py:31: in <module>
    import gevent.coros
E   ImportError: No module named coros
```

(Running python-2.7 tests on this branch to demonstrate - https://github.com/ramshaw888/apm-agent-python/commits/broken/zerorpc-tests )

`gevent.coros` was removed in v1.2.0, so I've pinned the version required for these tests to <1.2.0


Additionally, this moves the `zerorpc` tests out into a separate test env  (not sure of the nomenclature) `zerorpc-0.4`.

beniwohli commented:
Awesome stuff @ramshaw888! This also underlines why we need our coverage report situation fixed (we do report coverage to codecov.io, but for some reason it stopped reporting back coverage statistics).

ramshaw888 commented:
Few assumptions/notes I've made here:
* I've replaced the `sys.pypy_version_info < (2, 6)` check with `sys.version_info >= (3, 4)` since 2.7 is the lowest supported python version.
* From digging a little into git history I see a few `zerorpc` compatibility commits that  py version (and `zerorpc>=0.4.0,<0.5`) so I'll leave it as is for the purpose of this PR.
* I've added the `zerorpc-0.4` framework to `.jenkins_framework.yml` and `.jenkins_framework_full.yml`, any opinions on that?

beniwohli commented on tests/contrib/zerorpc/zeropc_tests.py:
```diff
@@ -36,16 +36,19 @@
-zerorpc = pytest.importorskip("zerorpc")
-gevent = pytest.importorskip("gevent")
 
+pytestmark = [
```
Nice! I never figured out how to have multiple module-level marks! This will come in handy :D

ramshaw888 commented:
I actually came across this when trying to improve the unit tests for https://github.com/elastic/apm-agent-python/pull/579 - I'd like to add another test framework environment for `gevent`, which requires pulling `gevent==1.2.2` from `requirements-base.txt`.

beniwohli commented on tests/contrib/zerorpc/zeropc_tests.py:
```diff
@@ -36,16 +36,19 @@
 
-zerorpc = pytest.importorskip("zerorpc")
-gevent = pytest.importorskip("gevent")
 
```
looks like isort doesn't like the two empty lines here, one should suffice :)

ramshaw888 replied:
gotcha (sorry forgot to install the precommit)

beniwohli replied:
No worries :) I'm mulling over how to make linting/autoformatting errors more visible anyway (see https://twitter.com/beniwohli/status/1168511125949943814 :D)
